plugin/pkg/admission/security/podsecuritypolicy/admission.go

```diff
@@ -102,6 +109,16 @@ func (a *podSecurityPolicyPlugin) SetInternalKubeInformerFactory(f informers.Sha
 	a.SetReadyFunc(podSecurityPolicyInformer.Informer().HasSynced)
 }
 
+func (c *podSecurityPolicyPlugin) Admit(a admission.Attributes) error {
+	return c.metrics.Admit(a)
```

Comment:
interesting wrapping pattern. Isn't it much easier and clean to just wrap in the `Register` at the top?

Reply:
Yes, but that breaks the `Wants*` interface pattern. I could just plumb the `SetAuthorizer` and `SetInternalKubeInformerFactory` methods through, but this approach required fewer changes to the metrics wrapper.

Reply:
that would be ugly as well, I agree.

In 1.9+master we get around this by adding the metric wiring to the chain setup code. Then the plugins are already initialized and the whole code is pretty clean. I have not looked how this would look like in 1.8. Have you?